### PR TITLE
MCFE-1135: Fix textarea initial height

### PR DESCRIPTION
**Pull request by chrischalstrom**
## Description
This fixes text area's initial height.  Previously if you had several lines worth of content, the height on the initial mount was still only one line.

## Slight issue
This change deleted the jsx snapshots for text areas because in order to make the image snapshotting work properly, I had to delay mounting the text area.  I found that the component mounts before css is loaded, measures itself, and then css loads resulting in an incorrect initial measurement.  This problem only seems to apply to the storybook environment - I tested this text area code in MC and it worked without any error.

I can edit the story to get correct image snapshots or correct jsx snapshots but doing both is very difficult because the image snapshot must wait until css has finished loading, while the jsx snapshot expects an immediate result.

## Removes console warnings
This change also removes console warnings seen when using textareas which include a warning about maxHeight, and a warning about uncontrolled inputs.

## Merge Checklist
- [x] Includes link to associated Github Issue or JIRA ticket
https://jira.sendgrid.net/browse/MCFE-1135
- [x] Has descriptive PR title and body
- [x] All changes are backwards compatible
- [x] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: src/text-area.tsx [hunk 1]
```diff
@@ -73,10 +80,12 @@ export class TextArea extends Component<
       value,
       tooltip,
       tooltipDirection,
+
+      // attributes are added to dom nodes and maxHeight is invalid, resulting in a console warning.
+      maxHeight,
```
Comment (mmissey) [inline src/text-area.tsx]: 👍

File: stories/TextArea.stories.tsx [hunk 2]
```diff
@@ -13,24 +13,34 @@ type RenderProp = (
...
   public handleChange: InputChangeEventHandler = (event: any) => {
     const target = event.target;
     this.setState({ value: target.value });
   };
 
+  public componentDidMount() {
+    // There is a really odd thing happening in storybook where it looks like without this, the text area mounts
+    // before css is applied, so when it measures itself to calculate its height, the measurement is off.
+    // This shouldn't be an issue in a normal situation where css loads before the page mounts.
+    setTimeout(() => this.setState({ ready: true }));
```
Comment (GiantRobots) [inline stories/TextArea.stories.tsx]: oh my god this is hacky af

Comment (chrischalstrom) [reply stories/TextArea.stories.tsx]: it's not great.  since this is just the story i don't super duper care a ton but ...

**Comment by thinkingserious** [comment]
Hello @chrischalstrom,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**